plugins/automox/icon_automox/util/api_client.py

```diff
@@ -112,12 +112,26 @@ def _page_results_data(self, url: str, params=None) -> List[Dict]:
     # Remove Null from response to avoid type issues
     def remove_null_values(self, d: Collection):
         if isinstance(d, dict):
-            return dict((k, self.remove_null_values(v)) for k, v in d.items() if v and self.remove_null_values(v))
+            return dict((k, self.remove_null_values(v)) for k, v in d.items() if self._is_value_or_number(v) and
+                        self._is_value_or_number(self.remove_null_values(v)))
         elif isinstance(d, list):
-            return [self.remove_null_values(v) for v in d if v and self.remove_null_values(v)]
+            return [self.remove_null_values(v) for v in d if self._is_value_or_number(v) and
+                    self._is_value_or_number(self.remove_null_values(v))]
         else:
             return d
 
+    @staticmethod
+    def _is_value_or_number(v) -> bool:
+        """
+        Check if value is not null or number to prevent 0 from being removed
+        """
+        if isinstance(v, (int, float)):
+            return True
+        elif v is not None:
+            return True
+        else:
+            return False
```

Comment:
```suggestion
    @staticmethod
    def _is_value_or_number(value: Any) -> bool:
        """
        Check if value is not null or number to prevent 0 from being removed
        """
        if isinstance(value, (int, float)):
            return True
        elif value is not None:
            return True
        else:
            return False
```